### PR TITLE
CORE-1244 chunking

### PR DESCRIPTION
**Pull request by rabbitonweb**
## Overview
* Actually chunk data
* Bring default max message size back to 4mb
* Keep max message size in range [1, 10] mb

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
CORE-1244

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/configuration/Configuration.scala [hunk 1]
```diff
@@ -57,7 +57,9 @@ object Configuration {
   private val DefaultRequiredSigns              = 0
   private val DefaultApprovalProtocolDuration   = 5.minutes
   private val DefaultApprovalProtocolInterval   = 5.seconds
-  private val DefaultMaxMessageSize: Int        = 100 * 1024 * 1024
+  private val DefaultMaxMessageSize: Int        = 4 * 1024 * 1024
```
Comment (sebtomba) [inline node/src/main/scala/coop/rchain/node/configuration/Configuration.scala]: Great!

File: node/src/main/scala/coop/rchain/node/configuration/Configuration.scala [hunk 2]
```diff
@@ -57,7 +57,9 @@ object Configuration {
...
+  private val MaxMessageSizeMinimumValue: Int   = 1 * 1024 * 1024
```
Comment (sebtomba) [inline node/src/main/scala/coop/rchain/node/configuration/Configuration.scala]: HTTP2 RFC 7540 specifies 2^14 or 16,384 octets as a minimum value. But let's stick with 1MB.

File: node/src/main/scala/coop/rchain/node/configuration/Configuration.scala [hunk 3]
```diff
@@ -57,7 +57,9 @@ object Configuration {
...
+  private val MaxMessageSizeMaximumValue: Int   = 10 * 1024 * 1024
```
Comment (sebtomba) [inline node/src/main/scala/coop/rchain/node/configuration/Configuration.scala]: Should be 2^24-1 or 16,777,215 octets like specified in the HTTP2 RFC 7540

Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/configuration/Configuration.scala]: I made it smaller on purpose, you want ot make it 16 mb?

File: comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala [hunk 4]
```diff
@@ -120,14 +120,15 @@ class TcpTransportLayer(host: String, port: Int, cert: String, key: String, maxM
...
     def header: Chunk =
       Chunk().withHeader(
         ChunkHeader().withSender(ProtocolHelper.node(blob.sender)).withTypeId(blob.packet.typeId)
       )
-
-    // FIX-ME actually chunk the data
-    def data: List[Chunk] = List(Chunk().withData(ChunkData().withContentData(blob.packet.content)))
+    val buffer = 2 * 1024 // 2 kbytes for protobuf related stuff
+    def data: List[Chunk] =
+      blob.packet.content.toByteArray.sliding(maxMessageSize - buffer, 1).toList.map { data =>
```
[the remark this answers is not in the document]
Comment (sebtomba) [reply comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: Please don't use `toList`! 🌴

[the remark this answers is not in the document]
Comment (rabbitonweb) [reply comm/src/main/scala/coop/rchain/comm/transport/TcpTransportLayer.scala]: 👍 🤦‍♂️

**Pull request merged**